src/query/ts/m3db/encoded_step_iterator.go

```diff
@@ -26,43 +26,70 @@ import (
 )
 
 type encodedStepIter struct {
-	consolidator *consolidators.StepLookbackConsolidator
+	collectors []*consolidators.StepLookbackConsolidator
 	encodedStepIterWithCollector
+	values []float64
 }
 
 func (b *encodedBlock) StepIter() (
 	block.StepIter,
 	error,
 ) {
-	cs := b.consolidation
-	iters := b.seriesBlockIterators
-	consolidator := consolidators.NewStepLookbackConsolidator(
-		b.lookback,
-		cs.bounds.StepSize,
-		cs.currentTime,
-		len(iters),
-		cs.consolidationFn,
+	var (
+		cs       = b.consolidation
+		iters    = b.seriesBlockIterators
+		lookback = b.options.LookbackDuration()
+
+		collectors       = make([]*consolidators.StepLookbackConsolidator, len(iters))
+		seriesCollectors = make([]consolidators.StepCollector, 0, len(iters))
```

Comment:
nit: Maybe make them both size zero but cap len(iters) or the opposite so it's consistent?

Reply:
Sure, is there any preference either way? I've generally leaned toward initializing explicitly with size when I know what the size will be, but not sure that's best (seems `append` is often a little safer)

Reply:
Append just tends to be safer yeah, I've seen a lot of bugs of creating size, then using append and then empty values sit in front of real values in the result.

If it's super straight forward and in super hot path it's probably justified to not use append (since might be slightly faster, don't need to keep updating the size of the array in the slice, although that's obviously super fast addition too).